utils/githooks/submodule_update_check.py

```diff
@@ -0,0 +1,51 @@
+"""Adds a warning message to the commit if a module is updated"""
+import re
+import subprocess
+import sys
+
+modified_re = re.compile(r'^(?:M|A)(\s+)(?P<name>.*)')
```

Comment:
Just FYI that python compiles and caches regex the first time it's used. So compiling upfront doesn't really help performance unless you have a lot (I think hundreds) of expressions